tests/php/tests/includes/test_class.access-token.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace WP_Job_Manager;
+
+use ReflectionObject;
+use WP_UnitTestCase;
+
+class WP_Test_Access_Token extends WP_UnitTestCase {
+
+	public function test_correct_token_verified_succesfully() {
+		$access_token = new Access_Token( [ 'user_id' => 10 ] );
+		$token = $access_token->create();
+
+		self::assertTrue( ( new Access_Token( [ 'user_id' => 10] ))->verify( $token, 1 ) );
+		self::assertFalse( ( new Access_Token( [ 'user_id' => 10] ))->verify( 'aRandomString', 1 ) );
+	}
+
+	public function test_expired_token_not_verified() {
+		$access_token = new Access_Token( [ 'user_id' => 10 ] );
+		$token = $access_token->create();
+
+		$generate_token = ( new ReflectionObject( $access_token ) )->getMethod( 'generate_token' );
+		$generate_token->setAccessible( true );
+		$tick = wp_nonce_tick();
+
+		$token_12_hours = $generate_token->invoke( $access_token, $tick - 1 );
+		$token_24_hours = $generate_token->invoke( $access_token, $tick - 2 );
+		$token_36_hours = $generate_token->invoke( $access_token, $tick - 3 );
+		$token_48_hours = $generate_token->invoke( $access_token, $tick - 4 );
+		$token_60_hours = $generate_token->invoke( $access_token, $tick - 5 );
+
+		self::assertTrue( ( new Access_Token( [ 'user_id' => 10] ))->verify( $token, 2 ) );
+		self::assertTrue( ( new Access_Token( [ 'user_id' => 10] ))->verify( $token_12_hours, 2 ) );
+		self::assertTrue( ( new Access_Token( [ 'user_id' => 10] ))->verify( $token_24_hours, 2 ) );
+		self::assertTrue( ( new Access_Token( [ 'user_id' => 10] ))->verify( $token_36_hours, 2 ) );
+		self::assertFalse( ( new Access_Token( [ 'user_id' => 10] ))->verify( $token_48_hours, 2 ) );
+		self::assertFalse( ( new Access_Token( [ 'user_id' => 10] ))->verify( $token_60_hours, 2 ) );
+	}
+
+	public function test_incorrect_payload_not_verified() {
+		$access_token = new Access_Token( [ 'user_id' => 10 ], time() + 1000 );
```

Comment:
I haven't run the tests to verify, but does this line actually work? Looks like it's handing in 2 args to the contructor, when it should only accept one.

Reply:
Yeah PHP ignores extra args. Fixed in 463a1e5be96967bfac0883e59b988974f28dd114